hsds/util/chunkUtil.py

```diff
@@ -1148,63 +1272,132 @@ def chunkQuery(
         raise ValueError(msg)
 
     if not slices:
-        slices = [
-            slice(0, dims[0], 1),
-        ]
+        slices = [slice(0, dims[0], 1), ]
     log.debug(f"chunkQuery slices: {slices}")
     if len(slices) != rank:
         msg = "Selection rank does not match shape rank"
         log.error(msg)
         raise ValueError(msg)
     slices = tuple(slices)
+    chunk_sel = chunk_arr[slices]
 
     chunk_coord = getChunkCoordinate(chunk_id, chunk_layout)
 
     # do query selection
     field_names = dset_dt.names
 
+    # get the eval str
+    eval_str = _getEvalStr(query, "chunk_sel", field_names)
+    if eval_str:
+        log.debug(f"eval_str: {eval_str}")
+    else:
+        log.debug("no eval_str")
+
+    # check for a where in statement
+    where_field = _getWhereFieldName(query)
+    if where_field:
+        log.debug(f"where_field: {where_field}")
+        if where_field not in field_names:
+            msg = f"where field {where_field} is not a member of dataset type"
+            raise ValueError(msg)
+        where_elements = _getWhereElements(query)
+        if not where_elements:
+            msg = "query: where key word with no elements"
+            raise ValueError(msg)
+        # convert to ndarray, checking that we can convert to our dtype along the way
+        try:
+            where_elements_arr = np.array(where_elements, dtype=dset_dt[where_field])
+        except ValueError:
+            msg = "where elements are not compatible with field datatype"
+            raise ValueError(msg)
+        isin_mask = np.isin(chunk_sel[where_field], where_elements_arr)
+
+        if not np.any(isin_mask):
+            # all false
+            log.debug("query - no rows found for where elements")
+            return None
+
+        isin_indices = np.where(isin_mask)
+        if not isinstance(isin_indices, tuple):
+            log.warn(f"expected where_indices of tuple but got: {type(isin_indices)}")
+            return None
+        if len(isin_indices) == 0:
+            log.warn("chunkQuery - got empty tuple where in result")
+            return None
+
+        isin_indices = isin_indices[0]
+        if not isinstance(isin_indices, np.ndarray):
+            log.warn(f"expected isin_indices of ndarray but got: {type(isin_indices)}")
+            return None
+        nrows = isin_indices.shape[0]
```

Comment:
This value doesn't seem to ever be used, since it's always overwritten by the assignment in line 1370